core/templates/pages/diagnostic-test-player-page/diagnostic-test-player.component.ts

```diff
@@ -29,19 +33,82 @@ import { PreventPageUnloadEventService } from 'services/prevent-page-unload-even
   OPPIA_AVATAR_IMAGE_URL: string = '';
-  diagnosticTestStarted = false;
+  diagnosticTestTopicTrackerModel!: DiagnosticTestTopicTrackerModel;
+  diagnosticTestIsStarted: boolean = false;
+  diagnosticTestIsFinished = false;
+  classroomUrlFragment: string = 'math';
+  recommendedTopicSummaries: CreatorTopicSummary[] = [];
+  recommendedTopicIds: string[] = [];
+  progressPercentage: number = 0;
+  componentSubscription = new Subscription();
+  classroomId: string = '';
 
   constructor(
     private urlInterpolationService: UrlInterpolationService,
-    private windowRef: WindowRef,
-    private preventPageUnloadEventService: PreventPageUnloadEventService
+    private preventPageUnloadEventService: PreventPageUnloadEventService,
+    private classroomBackendApiService: ClassroomBackendApiService,
+    private diagnosticTestPlayerStatusService: DiagnosticTestPlayerStatusService
   ) {}
 
   ngOnInit(): void {
-    this.preventPageUnloadEventService.addListener();
+    this.preventPageUnloadEventService.addListener(() => {
+      return (this.diagnosticTestIsStarted && !this.diagnosticTestIsFinished);
+    });
+
     this.OPPIA_AVATAR_IMAGE_URL = (
       this.urlInterpolationService.getStaticImageUrl(
         '/avatar/oppia_avatar_100px.svg'));
+
+    this.componentSubscription.add(
+      this.diagnosticTestPlayerStatusService
+        .onDiagnosticTestSessionCompleted.subscribe(
+          (recommendedTopicIds: string[]) => {
+            this.getRecommendedTopicSummaries(recommendedTopicIds);
+          }
+        ));
+
+    this.componentSubscription.add(
+      this.diagnosticTestPlayerStatusService
+        .onDiagnosticTestSessionProgressChange.subscribe(
+          (progressPercentage: number) => {
+            this.progressPercentage = progressPercentage;
+          }
+        ));
+    this.classroomBackendApiService.getClassroomIdAsync(
+      this.classroomUrlFragment).then(classroomId => {
+      this.classroomId = classroomId;
+    });
+  }
+
+  startDiagnosticTest(): void {
+    this.classroomBackendApiService.getClassroomDataAsync(
+      this.classroomId).then(response => {
+      this.diagnosticTestTopicTrackerModel = (
+        new DiagnosticTestTopicTrackerModel(
+          response.classroomDict.topicIdToPrerequisiteTopicIds));
+      this.diagnosticTestIsStarted = true;
+    });
+  }
+
+  getRecommendedTopicSummaries(recommendedTopicIds: string[]): void {
+    this.classroomBackendApiService.fetchClassroomDataAsync(
+      this.classroomUrlFragment).then((classroomData) => {
+      let topicSummaries: CreatorTopicSummary[] = (
+        classroomData.getTopicSummaries());
+      this.recommendedTopicSummaries = topicSummaries.filter(
+        (topicSummary) => {
+          return recommendedTopicIds.indexOf(topicSummary.getId()) !== -1;
+        });
+      this.diagnosticTestIsFinished = true;
+    });
+  }
+
+  getTopicButtonText(topicName: string): string {
+    return 'Start ' + topicName;
```

Comment:
This should be translated/internationalized.

Reply:
Done